ShinyExpenseTracker/src/ca/ualberta/cs/shinyexpensetracker/TagController.java

```diff
@@ -55,30 +57,40 @@ public static TagController getInstance() {
 		}
 	}
 	
-	public void setTagList(TagList list) {
-		this.list = list;
-	}
 
 	/**
 	 * Add a tag to the current tag list
 	 * <p>
-	 * 
-	 * @param s the string value of the added tag
+	 * @param tagString the string value of the added tag
+	 * @return boolean stating if the tag was added
 	 */
-	public boolean addTag(String s) {
-		if (s == null || s.equals("")){
+	public boolean addTag(String tagString) {
+		if(inTagList(tagString)){
+			return false;
+		}
+		if (tagString == null || tagString.equals("")){
 			return false;
 		}
 		Pattern p = Pattern.compile("^\\w*$");
-		Matcher m = p.matcher(s);
+		Matcher m = p.matcher(tagString);
 		if (m.matches()) {
-			list.addTag(s);
+			list.addTag(tagString);
 		return true;
 		}
 		else{
 			return false;
 		}
 	}
```

Comment:
I don't see it being reasonable to do the check in the tag class (e.g throwing an error in the constructor) but I can see the use of moving into tagList